tests/layering/layering_test.go

```diff
@@ -0,0 +1,207 @@
+	deployments []Deployments
+}
+
+func TestBootInClusterImage(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// create a new image stream
+	ctx := context.TODO()
+	imageStreamConfig := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: mcoNamespace,
+		},
+	}
+	_, err := cs.ImageStreams(mcoNamespace).Create(ctx, imageStreamConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	defer cs.ImageStreams(mcoNamespace).Delete(ctx, imageStreamName, metav1.DeleteOptions{})
+
+	// push a build to the image stream
+	buildConfig := &buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildName,
+		},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: "Git",
+					Git: &buildv1.GitBuildSource{
+						URI: "https://github.com/coreos/fcos-derivation-example",
+						Ref: "rhcos",
+					},
+				},
+				Strategy: buildv1.BuildStrategy{
+					DockerStrategy: &buildv1.DockerBuildStrategy{},
+				},
+				Output: buildv1.BuildOutput{
+					To: &v1.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: imageStreamName + ":latest",
+					},
+				},
+			},
+		},
+	}
+	_, err = cs.BuildV1Interface.Builds(mcoNamespace).Create(ctx, buildConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	defer cs.BuildV1Interface.Builds(mcoNamespace).Delete(ctx, buildName, metav1.DeleteOptions{})
+	waitForBuild(t, cs, buildConfig.ObjectMeta.Name)
+
+	// pick a random worker node
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/infra")
+	defer unlabelFunc()
+	infraNode := helpers.GetSingleNodeByRole(t, cs, "infra")
+
+	// get ImagePullSecret for the MCD service account and save to authfilePath on the node
+	// eventually we should use rest.InClusterConfig() instead of cs with kubeadmin
+	mcdServiceAccount, err := cs.ServiceAccounts(mcoNamespace).Get(ctx, "machine-config-daemon", metav1.GetOptions{})
+	require.Nil(t, err)
+	require.Equal(t, 1, len(mcdServiceAccount.ImagePullSecrets))
+	imagePullSecret, err := cs.Secrets(mcoNamespace).Get(ctx, mcdServiceAccount.ImagePullSecrets[0].Name, metav1.GetOptions{})
+	dockerConfigData := imagePullSecret.Data[corev1.DockerConfigKey]
+	var dockerConfig credentialprovider.DockerConfig
+	err = json.Unmarshal(dockerConfigData, &dockerConfig)
+	require.Nil(t, err)
+	dockerConfigJSON := credentialprovider.DockerConfigJSON{
+		Auths: dockerConfig,
+	}
+	authfileData, err := json.Marshal(dockerConfigJSON)
+	require.Nil(t, err)
+	helpers.ExecCmdOnNode(t, cs, infraNode, "mkdir", "-p", path.Dir(path.Join("/rootfs", authfilePath)))
+	// will get cleaned up on reboot since file is in /run
+	writeToMCDContainer(t, cs, infraNode, path.Join("/rootfs", authfilePath), authfileData)
+
+	// rpm-ostree rebase --experimental ostree-unverified-image:docker://image-registry.openshift-image-registry.svc.cluster.local:5000/openshift-machine-config-operator/test-boot-in-cluster-image-build
```

Comment:
```suggestion
	// rpm-ostree rebase --experimental ostree-unverified-image:image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/test-boot-in-cluster-image
```
or maybe just
```suggestion
	// rpm-ostree rebase --experimental imageURL
```
to avoid similar inconsistencies in the future

Reply:
No worries! I'm cleaning this up as I go.